OpenRA.Mods.Common/Traits/Armament.cs

```diff
@@ -118,7 +118,7 @@ public class Armament : PausableConditionalTrait<ArmamentInfo>, ITick
 
 		BodyOrientation coords;
 		INotifyBurstComplete[] notifyBurstComplete;
-		INotifyAttack[] notifyAttacks;
+		(Actor NotifyActor, INotifyAttack Notify)[] notifyAttacks;
```

Comment:
this is basically a static list, it will change incredibly rarely and for very few units. I thought this was more appropriate. And by this point to me it feels like normal code rather than gymnastics 😅

Reply:
Maybe it is better to use 
```List<(Actor NotifyActor, INotifyAttack[] Notifies)> notifyAttacks;```, which will save us some perf when adding or removing, as well as faster on searching.

> this is basically a static list, it will change incredibly rarely and for very few units.

Passing `notifyAttacks` between different actors can be a hot function for third party modders in their down stream code, we need some perfs here.

Reply:
it wouldn't happen often enough in third part mods as well. I changed it to use a list, and it's not really much cleaner